### PR TITLE
Option array of alternating colors for rainbow()

### PR DESCRIPTION
**Pull request by ir0ny1**
Become why not

### CODE REVIEW

**Comment by fent** [comment]
This is a good idea. Can you match the spacing style so that diff doesn't reflect unnecessary changes? And can you match the function signature comment style for consistency too?

**Comment by ir0ny1** [comment]
Should be good

File: lib/irc-colors.js [hunk 1]
```diff
@@ -67,18 +67,20 @@ Hash(styles).forEach(function(style, code) {
...
-  return str
-    .split('')
-    .map(function(c) {
-      return c !== ' ' ? exports[rainbow[i++ % l]](c) : c;
-    })
-    .join('');
+
+// Generate a rainbow. Optional array of colors as a second param
+exports.rainbow = function (str, clrArray) {
+    if (!clrArray)
```
Comment (fent) [inline lib/irc-colors.js]: can you add brackets around this if statement, for consistency

**Comment by fent** [comment]
The rest of the code is indented by 2 spaces, but in your change, the function indent has changed, even though you've only changed part of the function.

Can you also collapse your commits so the history doesn't look too weird.

**Pull request closed without merging**